nexus/db-model/src/db_metadata.rs

```diff
@@ -3,25 +3,35 @@
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
 use crate::schema::db_metadata;
+use crate::SemverVersion;
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
 
 /// Internal database metadata
-#[derive(Queryable, Insertable, Debug, Clone, Selectable)]
+#[derive(
+    Queryable, Insertable, Debug, Clone, Selectable, Serialize, Deserialize,
+)]
 #[diesel(table_name = db_metadata)]
 pub struct DbMetadata {
-    name: String,
```

Comment:
I did try keeping these values as strings, but there are major downsides:

- Parsing is a pain-in-the-butt
- Comparing the diff of built-in data between schema changes is terrible, in part because we can't identify "date" fields as such, and treat them as strings. As a consequence, `NOW() as TEXT` shows up as two completely different values when created at two different times.
- We could still have schema changes that modify strings! The changes are just implicit, and not codified in SQL itself

So:

I hear you that "if we want to use metadata as a part of schema updates, we need to update first". Totally agreed! But I think that's true of strings **and** of more strongly-typed schemas.